Semantics.fs

```diff
@@ -189,26 +185,29 @@ let markWrite (var : CTyped<Sym<Var>>) (idxPath : IntExpr<Sym<Var>> list)
 ///     Tries to extract the variable and index path from a lvalue.
 /// </summary>
 let varAndIdxPath (expr : Expr<Sym<Var>>)
-  : (CTyped<Sym<Var>> * IntExpr<Sym<Var>> list) option =
+  : (TypedVar * IntExpr<Sym<Var>> list) option =
     // TODO(CaptainHayashi): proper doc comment.
     // TODO(CaptainHayashi): merge with type lookup stuff in Modeller?
+    // TODO(CaptainHayashi): error perhaps if given a non-lvalue
 
     let rec getInBool bx path =
         match bx with
-        | BVar v -> Some (Bool v, path)
+        | BVar (Reg v) -> Some (Bool v, path)
+        // Symbols are not lvalues, so we can't process them.
```

Comment:
This is loosely enforced by the syntax forbidding lvalues, but a proper check might be better